bycycle/group/features.py

```diff
@@ -0,0 +1,276 @@
+
+###################################################################################################
+###################################################################################################
+
+def compute_features_2d(sigs, fs, f_range, compute_features_kwargs=None,
+                        return_samples=True, n_jobs=-1, progress=None):
+    """Compute shape and burst features for a 2 dimensional array of signals.
+
+    Parameters
+    ----------
+    sigs : 2d array
+        Voltage time series, with shape [n_signals, n_points].
+    fs : float
+        Sampling rate, in Hz.
+    f_range : tuple of (float, float)
+        Frequency range for narrowband signal of interest, in Hz.
+    compute_features_kwargs : dict or list of dict
+        Keyword arguments used in :func:`~.compute_features`.
+    return_samples : bool, optional, default: True
+        Whether to return a dataframe of cyclepoint sample indices.
+    n_jobs : int, optional, default: -1
+        The number of jobs, one per cpu, to compute features in parallel.
+    progress: {None, 'tqdm', 'tqdm.notebook'}, optional, default: None
+        Specify whether to display a progress bar. Use 'tqdm' if installed.
+
+    Returns
+    -------
+    df_features : list of pandas.DataFrame
+        Dataframes containing shape and burst features for each cycle.
+        Each dataframe is computed using the :func:`~.compute_features` function.
+    df_samples : list of pandas.DataFrame, optional
+        Dataframes containing cyclepoints for each cycle.
+        Only returned if ``return_samples`` is True.
+        Each dataframe is computed using the :func:`~.compute_features` function.
+
+    Notes
+    -----
+
+    - The order of ``df_features`` and ``df_samples`` corresponds to the order of ``sigs``.
+    - If ``compute_features_kwargs`` is a dictionary, the same kwargs are applied applied across
+      the first axis of ``sigs``. Otherwise, a list of dictionaries equal in length to the
+      first axis of ``sigs`` is required to apply unique kwargs to each signal.
+    - ``return_samples`` is controlled from the kwargs passed in this function. If
+      ``return_samples`` is a key in ``compute_features_kwargs``, it's value will be ignored.
+
+    Examples
+    --------
+    Compute the features of a 2d array in parrallel using the same compute_features kwargs.
+
+    >>> import numpy as np
+    >>> from neurodsp.sim import sim_bursty_oscillation
+    >>> fs, f_range = 500, (8, 12)
+    >>> sigs = np.array([sim_bursty_oscillation(10, fs, 10) for i in range(10)])
+    >>> compute_kwargs = {'burst_method': 'amp', 'threshold_kwargs':{'burst_fraction_threshold': 1}}
+    >>> df_features = compute_features_2d(sigs, fs, f_range, return_samples=False, n_jobs=2,
+    ...                                   compute_features_kwargs=compute_kwargs)
+
+    Compute the features of a 2d array in parallel using unique compute_features kwargs to examine
```

Comment:
I'm not sure `unique compute_features kwargs` is particularly obvious, so maybe "using individualized settings per signal" or something similar.